### PR TITLE
Overview documentation

### PR DESCRIPTION
**Pull request by lkpttn**
### What does this PR do?
Adds the Philosophical Overview to the docmentation

### Any background context you want to provide?
It's not complete, but everyone can look at it now!

### CODE REVIEW

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -20,6 +20,16 @@
...
+- [Overview](philosophical-overview/overview/overview.md)
+  + [Goals](philosophical-overview/overview/goals.md)
+  + [Pages](philosophical-overview/overview/overview.md#pages)
+  + [Components](philosophical-overview/overview/overview.md#components)
+  + [Tabs](philosophical-overview/overview/overview.md#tabs)
+  + [Blocks](philosophical-overview/overview/overview.md#blocks)
+  + [Actions](philosophical-overview/overview/overview.md#actions)
+  + [Contentful](philosophical-overview/overview/overview.md#contentful)
+  + [Sharing](philosophical-overview/overview/overview.md#sharing)
+  + [Decisions](philosophical-overview/overview/decisions.md)
```
Comment (weerd) [inline docs/SUMMARY.md]: Should the `Decisions` item still be a child of Overview or at the same level?

File: docs/philosophical-overview/Overview/overview.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
+
+# Phoenix-Next Philosophical Overview
```
Comment (weerd) [inline docs/philosophical-overview/Overview/overview.md]: `Phoenix (next)`  <-- change it to how it is in the rest of the docs 😘

**Comment by weerd** [comment]
I kind of feel you could simplify the structure, get rid of the nested overview directory and separate each section into their own pages so you have shorter, easier to read pages:

```
## Philosophical Overview
- [Introduction](philosophical-overview/README.md)
- [Goals](philosophical-overview/goals.md)
- [Pages](philosophical-overview/pages.md)
- [Components](philosophical-overview/components.md)
- [Tabs](philosophical-overview/tabs.md)
- [Blocks](philosophical-overview/blocks.md)
- [Actions](philosophical-overview/actions.md)
- [Contentful](philosophical-overview/contentful.md)
- [Sharing](philosophical-overview/sharing.md)
- [Decisions](philosophical-overview/decisions.md)
```

The intro in `philosophical-overview/README.md` could also house a simple ToC for the overview like I did in some of the other introductions in other sections.

**Comment by lkpttn** [comment]
Breaking everything up into lots of mini-docs seems inconvenient. Especially when something needs to be updated and I have to open 5 or 6 files to reflect the changes.

**Comment by itsjoekent** [comment]
Yeah, I don't think Luke has to go overkill with making markdown files and spending a ton of time on this. If we start using this document frequently and run into navigation issues then we can add a table of contents or something simple.

**Comment by weerd** [comment]
```
## Philosophical Overview
- [Introduction](philosophical-overview/README.md)
- [Overview](philosophical-overview/overview/README.md)
  + [Pages](philosophical-overview/overview/README.md#pages)
  + [Components](philosophical-overview/overview/README.md#components)
  + [Tabs](philosophical-overview/overview/README.md#tabs)
  + [Blocks](philosophical-overview/overview/README.md#blocks)
  + [Actions](philosophical-overview/overview/README.md#actions)
  + [Contentful](philosophical-overview/overview/README.md#contentful)
  + [Sharing](philosophical-overview/overview/README.md#sharing)
- [Goals](philosophical-overview/overview/goals.md)
- [Decisions](philosophical-overview/overview/decisions.md)
```
Goals and Decisions shouldn't be children of Overview but siblings in the hierarchy since they are their own pages.

But yeah, whatever is easier just want an understandable hierarchy. Feel free to get rid of the introduction; mostly added those as placeholders and if they make sense for a section, keep it but can otherwise get rid of it.

Also if you have a folder, the `README.md` file in a folder behaves sorta like an `index.html` and will load as the main file if the url path is just to that folder.

**Comment by weerd** [comment]
@lkpttn for reference here's GitBook's own [SUMMARY.md](https://raw.githubusercontent.com/GitbookIO/gitbook/master/docs/SUMMARY.md) for their docs on how to use GitBooks.

**Pull request merged**